express-notes-api/index.js

```diff
@@ -0,0 +1,90 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+const fs = require('fs');
+
+app.use(express.json());
+
+const newJSON = JSON.stringify(data, null, 2);
+
+app.get('/api/notes', (req, res) => {
+  const array = [];
+  if(!data.notes){
+    return [];
+  }
+  for (const index in data.notes) {
+    array.push(data.notes[index]);
+  }
+   res.status(200).json(array);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  if (isNaN(req.params.id)){
```

Comment:
You also need to confirm that it is a positive integer. This is helpful for that: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger